tests/unittests/PartitionerTest.cpp

```diff
@@ -526,40 +526,91 @@ static void createSimpleModule(Module &mod) {
 }
 
 TEST_F(PartitionerTest, SimpleHeterogeneousPartitioning) {
-  {
-    createSimpleModule(mod_);
-    BackendWithoutSub backendWithoutSub1, backendWithoutSub2;
-    BackendWithoutMul backendWithoutMul1, backendWithoutMul2;
-    // Create two backends which support different ops, then do the partition by
-    // assigning the ops to the corresponding abackends.
-    std::vector<Backend *> backends;
-    backends.emplace_back(&backendWithoutMul1);
-    backends.emplace_back(&backendWithoutMul2);
-    backends.emplace_back(&backendWithoutSub1);
-    backends.emplace_back(&backendWithoutSub2);
-    std::vector<DeviceInfo> devices = {{3072, "Interpreter"},
-                                       {3072, "Interpreter"},
-                                       {3072, "CPU"},
-                                       {3072, "CPU"}};
-    auto partitioner =
-        Partitioner(&mod_, devices, backends, /* saturateHost */ true);
-    CompilationContext cctx;
-    auto err = partitioner.Partition(cctx);
-    EXPECT_FALSE(errToBool(std::move(err)));
-    DAGListTy myList = std::move(partitioner.getPartitionResult());
-    ASSERT_EQ(mod_.getFunctions().size(), 3);
-    ASSERT_EQ(myList.size(), 1);
-    ASSERT_TRUE(checkSaveNode(mod_));
-
-    for (auto &dag : myList) {
-      for (auto &node : dag.nodes) {
-        // Although the saturateHost is set true, no saturating the host in
-        // heterogeneous partiton.
-        ASSERT_EQ(node->logicalDevices.size(), 1);
-      }
+  createSimpleModule(mod_);
+  BackendWithoutSub backendWithoutSub1, backendWithoutSub2;
+  BackendWithoutMul backendWithoutMul1, backendWithoutMul2;
+  // Create two backends which support different ops, then do the partition by
+  // assigning the ops to the corresponding abackends.
+  std::vector<Backend *> backends;
+  backends.emplace_back(&backendWithoutMul1);
+  backends.emplace_back(&backendWithoutMul2);
+  backends.emplace_back(&backendWithoutSub1);
+  backends.emplace_back(&backendWithoutSub2);
+  std::vector<DeviceInfo> devices = {{3072, "Interpreter"},
+                                     {3072, "Interpreter"},
+                                     {3072, "CPU"},
+                                     {3072, "CPU"}};
+  auto partitioner =
+      Partitioner(&mod_, devices, backends, /* saturateHost */ true);
+  CompilationContext cctx;
+  auto err = partitioner.Partition(cctx);
+  EXPECT_FALSE(errToBool(std::move(err)));
+  DAGListTy myList = std::move(partitioner.getPartitionResult());
+  ASSERT_EQ(mod_.getFunctions().size(), 3);
+  ASSERT_EQ(myList.size(), 1);
+  ASSERT_TRUE(checkSaveNode(mod_));
+
+  for (auto &dag : myList) {
+    for (auto &node : dag.nodes) {
+      // Although the saturateHost is set true, no saturating the host in
+      // heterogeneous partiton.
+      ASSERT_EQ(node->logicalDevices.size(), 1);
+    }
+  }
+  mod_.clear();
+}
+
+/// Test pre-defined non-supported ops used for choosing backend in
+/// Heterogeneous Partition.
```

Comment:
It is not obvious to me from the code and comments below how the non-supported ops are tested.

Reply:
The config yaml file is only loaded in "loader" flow (I added resnet50 test in tests/images/run.sh to test loading and parsing the yaml file). For Partitioner unnitest, we just pass the pre-defined nodes list to DeviceInfo used in Partitioner. Added comments in the tests.